types/mithril/index.d.ts

```diff
@@ -246,7 +246,7 @@ declare namespace Mithril {
 	 * Components are a mechanism to encapsulate parts of a view to make code easier to organize and/or reuse.
 	 * Any Javascript object that has a view method is a Mithril component. Components can be consumed via the m() utility.
 	 */
-	type Comp<Attrs, State extends Lifecycle<Attrs, State>> = Component<Attrs, State> & State;
```

Comment:
@spacejack Thanks for the examples. The first one indeed incorrectly passes with this change. And I expected the second one to pass too; there just happen to be enough type checks throughout this package that it gets caught elsewhere.

I thought of a workaround that keeps the constraint and doesn't pollute `Lifecycle` too badly, I think: I added a string indexer `[s: string]: any` to it. This means that `Lifecycle` is no longer a weak type, but doesn't hurt assignability when used with `extends`. But it means that the uses in `Hyperscript` might be affected. I don't know if Hyperscript is something that users implement or whether they just use it.